### PR TITLE
build: use Makefile test command for CI tests

### PR DESCRIPTION
**Pull request by deadprogram**
This PR switches over to using a Makefile to run the CI tests. This has the additional benefit of making it easier to run the smoke test suite from the command line locally.

The intention is to also make it easier to run this same tests suite from the main TinyGo repo to ensure that none of the drivers/examples are broken by the ongoing development taking place.

### CODE REVIEW

**Comment by deadprogram** [comment]
Note that this PR is current failing CI tests because of https://github.com/tinygo-org/tinygo/issues/351 once that has been corrected the tests should pass.

**Comment by deadprogram** [comment]
Any opinions from anyone on this PR before I merge it in? We need this to keep all the code/tests in sync with the main repo.

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,30 +13,4 @@ jobs:
       - run:
           name: "Enforce Go Formatted Code"
           command: "! go fmt ./... 2>&1 | read"
```
Comment (aykevl) [inline .circleci/config.yml]: Any reason not to replace this with `make fmt-check`?

Comment (deadprogram) [reply .circleci/config.yml]: Ok done.

**Comment by aykevl** [comment]
LGTM, but should this PR be merged now or after https://github.com/tinygo-org/tinygo/issues/351 has been fixed?

**Comment by deadprogram** [comment]
It can be merged now, seems to me.
